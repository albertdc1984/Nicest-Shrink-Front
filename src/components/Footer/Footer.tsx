import styled from "styled-components";

const Futer = styled.footer`
  display: flex;
  position: fixed;
  width: 100vw;
  justify-content: center;
  background-color: #f37370;
  align-items: center;
  color: white;
  height: 15vh;
  bottom: 0;
  overflow: scroll;
  box-shadow: 0px 0px 10px white;
`;
const Image = styled.img`
  height: 90%;
`;

const Footer = (): JSX.Element => {
  return (
    <Futer>
      <Image src={require("../../logo.png")} alt="Nicest Shrink Logo" />
    </Futer>
  );
};

export default Footer;
