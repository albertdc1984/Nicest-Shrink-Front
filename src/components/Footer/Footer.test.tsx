import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router-dom";
import Footer from "./Footer";
import TestRenderer from "react-test-renderer";

describe("Given a Footer component", () => {
  describe("When it's instantiated", () => {
    test("Then it should render a logo that renders a menu when clicked", async () => {
      render(
        <MemoryRouter>
          <Footer />
        </MemoryRouter>
      );

      const logo = screen.getByTitle("onclick");
      const menu = await screen.findByTitle("menuList");
      await userEvent.click(logo);

      expect(logo).toBeInTheDocument();
      expect(menu).toBeInTheDocument();
    });
    test("Then it shouldn't render a list if the image has not been clicked", async () => {
      render(
        <MemoryRouter>
          <Footer />
        </MemoryRouter>
      );

      const logo = screen.getByTitle("onclick");
      await userEvent.click(logo);
      const menu = await screen.findByTitle("menuList");
      const link = await screen.findByTitle("calendar");
      await userEvent.click(link);

      expect(menu).not.toBeVisible();
    });
  });
  test("then it should match the snapshot", () => {
    const app = TestRenderer.create(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    ).toJSON();

    expect(app).toMatchSnapshot();
  });
});
