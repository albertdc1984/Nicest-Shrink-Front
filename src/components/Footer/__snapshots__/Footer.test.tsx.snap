// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Footer component then it should match the snapshot 1`] = `
<footer
  className="sc-hKwDye gEnURZ"
>
  <div
    className="sc-bdvvtL fdCMAt menu-container unactive"
    data-testid="onclick"
    onClick={[Function]}
    title="onclick"
  >
    <ul
      className="sc-gsDKAQ cdBSrP"
      title="menuList"
    >
      <li>
        <div
          className="sc-dkPtRN EebdW"
        >
          <a
            href="/calendar"
            onClick={[Function]}
            title="calendar"
          >
            Calendar
          </a>
        </div>
      </li>
      <li>
        <div
          className="sc-dkPtRN EebdW"
        >
          <a
            href="/sessions"
            onClick={[Function]}
          >
            Sessions
          </a>
        </div>
      </li>
      <li>
        <div
          className="sc-dkPtRN EebdW"
        >
          <a
            href="/patients"
            onClick={[Function]}
          >
            Patients
          </a>
        </div>
      </li>
      <li>
        <div
          className="sc-dkPtRN EebdW"
        >
          <a
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </li>
      <li>
        <div
          className="sc-dkPtRN EebdW"
        >
          <a
            href="/newsession"
            onClick={[Function]}
          >
            Create Session
          </a>
        </div>
      </li>
    </ul>
     
  </div>
  <h4
    className="sc-eCImPb kurszg"
  >
    MENU
  </h4>
  <img
    alt="Nicest Shrink Logo"
    className="sc-jRQBWg frKrjU"
    onClick={[Function]}
    src="logo.png"
  />
</footer>
`;
